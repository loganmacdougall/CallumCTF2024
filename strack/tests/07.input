# bottom of stack is i

# for (i = 1; i != 11; i++) {

0 1 ADD
11 2 DUP 1 POP EQ -99 CJUMP # Out-of-bound instruction - exit program

# if (i < 5) {

5 2 DUP 1 POP LESS NOT 10 CJUMP

# print(i, " is less than 5\n")

1 DUP INTSTRING PRINT " is less than 5\n" PRINT 1 -27 CJUMP

# } else if (i > 5) {

5 2 DUP 1 POP MORE NOT 10 CJUMP

# print(i, " is more than 5\n")

1 DUP INTSTRING PRINT " is more than 5\n" PRINT 1 -45 CJUMP

# } else print(i, " is equal to 5\n")

1 DUP INTSTRING PRINT " is equal to 5\n" PRINT 1 -54 CJUMP

# }