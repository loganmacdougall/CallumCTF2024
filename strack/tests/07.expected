1 is less than 5
2 is less than 5
3 is less than 5
4 is less than 5
5 is equal to 5
6 is more than 5
7 is more than 5
8 is more than 5
9 is more than 5
10 is more than 5
